theano/tensor/nnet/tests/test_spatialtf.py

```diff
@@ -0,0 +1,222 @@
+    def setUp(self):
+        utt.seed_rng()
+
+    def getInputs(self):
+        inp_shape = (np.random.randint(1, 11), 3, np.random.randint(10, 101),
+                     np.random.randint(10, 101))
+        num_images, num_channels, height, width = inp_shape
+
+        inp = np.random.random(inp_shape).astype(config.floatX)
+
+        transform = [[-1, 0, 0],
+                     [0, -1, 0]]
+        theta = np.asarray(num_images * [transform], dtype=config.floatX)
+        scale_height = np.random.random()
+        scale_width = np.random.random()
+
+        return (inp, theta, scale_width, scale_height)
+
+    def test_symb(self):
+        """
+        Compare CPU implementation with a symbolic one
+        """
+        inp, theta, scale_width, scale_height = self.getInputs()
+
+        # Setup symbolic variables
+        t_inp = T.tensor4('inp')
+        t_theta = T.tensor3('theta')
+        t_scale_height = T.scalar('scale_height')
+        t_scale_width = T.scalar('scalar_width')
+
+        symb_out_op = spatialtf_symb(t_inp, t_theta, t_scale_width, t_scale_height)
+        symb_out_fn = theano.function([t_inp, t_theta, t_scale_width, t_scale_height],
+                                      symb_out_op, mode=mode_without_gpu)
+        symb_out = symb_out_fn(inp, theta, scale_width, scale_height)
+
+        cpu_out_op = spatialtf(t_inp, t_theta, t_scale_width, t_scale_height)
+        cpu_out_fn = theano.function([t_inp, t_theta, t_scale_width, t_scale_height],
+                                     cpu_out_op, mode=mode_without_gpu)
+        cpu_out = cpu_out_fn(inp, theta, scale_width, scale_height)
+
+        # Check if results are approx. equal
+        utt.assert_allclose(symb_out, cpu_out)
+
+    def test_gpu(self):
+        inp, theta, scale_width, scale_height = self.getInputs()
+        # Setup symbolic variables
+        t_inp = T.tensor4('inp')
+        t_theta = T.tensor3('theta')
+        t_scale_height = T.scalar('scale_height')
+        t_scale_width = T.scalar('scalar_width')
+        # Setup spatial transformer on the GPU using cuDNN
+        op_gpu = dnn_spatialtf(t_inp, t_theta, t_scale_width, t_scale_height)
+        fn_gpu = theano.function([t_inp, t_theta, t_scale_width, t_scale_height],
+                                 op_gpu, mode=mode_with_gpu)
+        # Evaluate GPU function
+        out_gpu = fn_gpu(inp, theta, scale_width, scale_height)
+
+        for border_mode in ('nearest', 'mirror', 'wrap'):
+            # Setup spatial transformer on the CPU
+            op_cpu = spatialtf(t_inp, t_theta, t_scale_width, t_scale_height, border_mode)
+            fn_cpu = theano.function([t_inp, t_theta, t_scale_width, t_scale_height],
+                                     op_cpu, mode=mode_without_gpu)
+            # Evaluate CPU function
+            out_cpu = fn_cpu(inp, theta, scale_width, scale_height)
+
+            # Check results
+            utt.assert_allclose(out_gpu, out_cpu)
```

Comment:
`out_gpu` -> `np.asarray(out_gpu)`

Reply:
Addressed in commit 88efff8440cc492522799d2205e0bc9537780a2d